src/titiler/xarray/titiler/xarray/io.py

```diff
@@ -135,13 +74,21 @@ def xarray_open_dataset(  # noqa: C901
                 "lock": False,
             }
         )
-
-        ds = xarray.open_dataset(file_handler, **xr_open_args)
+        fs = fsspec.filesystem(protocol)
+        ds = xarray.open_dataset(fs.open(src_path), **xr_open_args)
 
     # Fallback to Zarr
     else:
-        ds = xarray.open_zarr(file_handler, **xr_open_args)
-
+        if _zarr_v3:
+            if protocol == "file":
+                store = zarr.storage.LocalStore(parsed.path, read_only=True)
+            else:
+                fs = fsspec.filesystem(protocol, storage_options={"asynchronous": True})
+                store = zarr.storage.FsspecStore(fs, path=src_path, read_only=True)
+            ds = xarray.open_zarr(store, **xr_open_args)
+        else:
+            fs = fsspec.filesystem(protocol)
+            ds = xarray.open_zarr(fs.get_mapper(src_path), **xr_open_args)
```

Comment:
This is the bulk of the changes to account for Zarr no longer accepting fsspec's mutable mapping wrappers and expecting asynchronous file systems.